tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -42,55 +57,2276 @@ def test_convert_labels_to_ids(domain: Domain):
+            },
+            {
+                USER: {INTENT: "mood_great"},
+                PREVIOUS_ACTION: {ACTION_NAME: "utter_happy"},
+            },
+            {
+                USER: {INTENT: "mood_great"},
+                PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+            },
+        ]
+    ]
+
+    if max_history is not None:
+        expected_states = [x[-max_history:] for x in expected_states]
+
+    assert actual_states is not None
+    assert len(actual_states) == len(expected_states)
+
+    for actual, expected in zip(actual_states, expected_states):
+        assert actual == expected
+
+
+@pytest.mark.parametrize("max_history", [None, 3])
+def test_prediction_states_ignores_action_intent_unlikely_with_max_history_tracker_featurizer(
+    moodbot_tracker: DialogueStateTracker,
+    moodbot_domain: Domain,
+    max_history: Optional[int],
+):
+
+    state_featurizer = SingleStateFeaturizer()
+    tracker_featurizer = MaxHistoryTrackerFeaturizer(
+        state_featurizer, max_history=max_history
+    )
+
+    tracker = DialogueStateTracker.from_events(
+        "default",
+        [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered("greet"),
+            ActionExecuted(ACTION_UNLIKELY_INTENT_NAME),
+            ActionExecuted("utter_greet"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered("mood_great"),
+            ActionExecuted(ACTION_UNLIKELY_INTENT_NAME),
+            ActionExecuted("utter_happy"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered("goodbye"),
+        ],
+        domain=moodbot_domain,
+    )
+
+    actual_states = tracker_featurizer.prediction_states(
+        [tracker], moodbot_domain, ignore_action_unlikely_intent=True
+    )
+
+    expected_states = [
+        [
+            {},
+            {
+                USER: {INTENT: "greet"},
+                PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+            },
+            {USER: {INTENT: "greet"}, PREVIOUS_ACTION: {ACTION_NAME: "utter_greet"},},
+            {
+                USER: {INTENT: "mood_great"},
+                PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+            },
+            {
+                USER: {INTENT: "mood_great"},
+                PREVIOUS_ACTION: {ACTION_NAME: "utter_happy"},
+            },
+            {
+                USER: {INTENT: "goodbye"},
+                PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+            },
+        ]
+    ]
+
+    if max_history is not None:
+        expected_states = [x[-max_history:] for x in expected_states]
+
+    assert actual_states is not None
+    assert len(actual_states) == len(expected_states)
+
+    for actual, expected in zip(actual_states, expected_states):
+        assert actual == expected
+
+
+@pytest.mark.parametrize("max_history", [None, 3])
+def test_prediction_states_keeps_action_intent_unlikely_with_max_history_tracker_featurizer(
```

Comment:
Over here too, you can use parameterization to get rid of redundant code. (Make `ignore_action_unlikely_intent` and expected states parameters for the tests.)